core/storage/base_model/gae_models.py

```diff
@@ -720,6 +720,73 @@ def delete(self, committer_id, commit_message, force_deletion=False):
                 committer_id, self._COMMIT_TYPE_DELETE, commit_message,
                 commit_cmds)
 
+    @classmethod
+    def delete_multi(cls, entity_ids, committer_id,
+                     commit_message, force_deletion=False):
+        """Deletes the given cls instancies with the given entity_ids.
+
+        Args:
+            entity_ids: list(str). Ids of entities to delete.
+            committer_id: str. The user_id of the user who committed the change.
+            commit_message: str.
+            force_deletion: bool. If True these models are deleted completely
+                from storage, otherwise there are only marked as deleted.
+                Default is False.
+
+        Raises:
+            Exception: This model instance has been already deleted.
+        """
+        versioned_models = cls.get_multi(entity_ids)
+        if force_deletion:
+            all_models_metadata_keys = []
+            all_models_content_keys = []
+            for model in versioned_models:
+                model_version_numbers = [
+                    python_utils.UNICODE(num + 1) for num in
+                    python_utils.RANGE(model.version)]
+                model_snapshot_ids = [
+                    model.get_snapshot_id(model.id, version_number)
+                    for version_number in model_version_numbers]
+
+                all_models_metadata_keys.extend([
+                    ndb.Key(model.SNAPSHOT_METADATA_CLASS, snapshot_id)
+                    for snapshot_id in model_snapshot_ids])
+                all_models_content_keys.extend([
+                    ndb.Key(model.SNAPSHOT_CONTENT_CLASS, snapshot_id)
+                    for snapshot_id in model_snapshot_ids])
+
+            ndb.delete_multi(
+                all_models_metadata_keys + all_models_content_keys)
+            super(VersionedModel, cls).delete_multi(versioned_models)
+        else:
+            for model in versioned_models:
+                model._require_not_marked_deleted()  # pylint: disable=protected-access
+                model.deleted = True
+
+            commit_cmds = [{
+                'cmd': cls.CMD_DELETE_COMMIT
+            }]
+            snapshot_metadata_models = []
+            snapshot_content_models = []
+            for model in versioned_models:
+                model.version += 1
+                snapshot = model.compute_snapshot()
+                snapshot_id = model.get_snapshot_id(model.id, model.version)
+
+                snapshot_metadata_models.append(model.SNAPSHOT_METADATA_CLASS(  # pylint: disable=not-callable
```

Comment:
This probably should be defined as a method on the metadata class, so that if the fields change, this also gets changed.

Ditto for SNAPSHOT_CONTENT_CLASS below.

Reply:
Done.